autonomic-administration-plugin/src/test/java/br/com/autonomiccs/autonomic/administration/plugin/AdministrationAgentTest.java

```diff
@@ -0,0 +1,665 @@
+import br.com.autonomiccs.autonomic.algorithms.commons.resources.HostResources;
+import br.com.autonomiccs.autonomic.algorithms.commons.resources.VmResources;
+import br.com.autonomiccs.autonomic.algorithms.commons.services.CloudResourcesService;
+import br.com.autonomiccs.autonomic.algorithms.commons.services.ClusterResourcesService;
+import br.com.autonomiccs.autonomic.algorithms.commons.services.HostResourcesService;
+import br.com.autonomiccs.autonomic.plugin.common.services.AutonomicClusterManagementHeuristicService;
+import br.com.autonomiccs.autonomic.plugin.common.services.ClusterService;
+import br.com.autonomiccs.autonomic.plugin.common.services.HostService;
+import br.com.autonomiccs.autonomic.plugin.common.services.VirtualMachineService;
+import br.com.autonomiccs.autonomic.plugin.common.services.ZoneService;
+import br.com.autonomiccs.autonomic.plugin.common.utils.ThreadUtils;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AdministrationAgentTest {
+
+    @Spy
+    @InjectMocks
+    private AdministrationAgent administrationAgent;
+    @Mock
+    private ClusterService clusterService;
+    @Mock
+    private AutonomicClusterManagementHeuristicService autonomicManagementHeuristicService;
+    @Mock
+    private ClusterAdministrationHeuristicAlgorithm clusterAdministrationHeuristicAlgorithm;
+    @Mock
+    private ZoneService zoneService;
+    @Mock
+    private AutonomicClusterManagementService autonomicClusterManagementService;
+    @Mock
+    private CloudResourcesService cloudResourcesService;
+    @Mock
+    private ClusterResourcesService clusterResourcesService;
+    @Mock
+    private HostService hostService;
+    @Mock
+    private VirtualMachineService virtualMachineService;
+    @Mock
+    private ThreadUtils threadUtils;
+    @Mock
+    private UserVmService userVmService;
+    @Mock
+    private HypervisorManager hypervisorManager;
+    @Mock
+    private HostResourcesService hostResourcesService;
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void receiveClusterToBeManagedTestClusterIdNull() {
+        administrationAgent.receiveClusterToBeManaged(null);
+        verifyReceiveClusterToBeManaged(0, 0);
+    }
+
+    @Test
+    public void receiveClusterToBeManagedTestClusterIdNotNullButClusterIsNull() {
+        administrationAgent.receiveClusterToBeManaged(1l);
+        verifyReceiveClusterToBeManaged(1, 0);
+    }
+
+    @Test
+    public void receiveClusterToBeManagedTestClusterIdAndClusterNotNull() {
+        ClusterVO cluster = Mockito.mock(ClusterVO.class);
+        Mockito.doReturn(cluster).when(clusterService).findById(1l);
+        Mockito.doNothing().when(administrationAgent).workOnCluster(Mockito.any(ClusterVO.class), Mockito.any(ClusterAdministrationHeuristicAlgorithm.class));
+
+        administrationAgent.receiveClusterToBeManaged(1l);
+        verifyReceiveClusterToBeManaged(1, 1);
```

Comment:
after this you should also check if "workOnCluster" was executed with the cluster and administrationAlgorithm returned